### PR TITLE
Enable linting rule MD032 (blank lines around lists)

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) enables rule MD032, which requires blank lines around lists.

Related to #5250 

## Additional info

This PR fixes a majority of issues, but there are still 38 errors:

```
rake test:md
Testing Markdown style with mdl ...
guides/v2.2/cloud/cdn/trouble-fastly.md:208: MD032 Lists should be surrounded by blank lines
guides/v2.2/cloud/cdn/trouble-fastly.md:237: MD032 Lists should be surrounded by blank lines
guides/v2.2/mrg/ce/Signifyd.md:48: MD032 Lists should be surrounded by blank lines
guides/v2.3/cloud/cdn/trouble-fastly.md:208: MD032 Lists should be surrounded by blank lines
guides/v2.3/cloud/cdn/trouble-fastly.md:237: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/b2b/CompanyCredit.md:15: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Analytics.md:38: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Catalog.md:8: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/CatalogWidget.md:10: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Deploy.md:16: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Deploy.md:17: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Indexer.md:10: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/InstantPurchase.md:12: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Paypal.md:8: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Reports.md:8: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Robots.md:8: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Sales.md:12: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Sales.md:15: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/SalesSequence.md:12: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/SalesSequence.md:15: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/SampleData.md:30: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/SampleData.md:64: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Security.md:11: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Signifyd.md:50: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Tax.md:12: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Tax.md:13: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Tax.md:17: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Tax.md:22: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Tax.md:24: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Weee.md:25: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ce/Weee.md:32: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ee/CatalogPermissions.md:9: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ee/CatalogPermissions.md:12: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ee/Staging.md:13: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ee/Staging.md:19: MD032 Lists should be surrounded by blank lines
guides/v2.3/mrg/ee/Staging.md:31: MD032 Lists should be surrounded by blank lines

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md

The Markdown linter has found 38 issues
```

The cloud errors are interesting because the reported issue is inside a fenced code block and is considered a false positive because it's intended to be terminal output.

The MRG errors must be fixed in the source READMEs; not in devdocs.

I propose reverting the change in this PR that enables the rule, but merging all the other fixes while we fix the MRG READMEs and address the outstanding cloud issue.

### CODE REVIEW

**Comment by jeff-matthews** [comment]
I turned off the rule. We can merge this PR pending approval.

File: guides/v2.3/config-guide/cli/config-cli-subcommands-index.md [hunk 1]
```diff
@@ -121,28 +121,7 @@ Catalog Search index has been rebuilt successfully in <time>
...
-For example, the following command runs the Catalog Search Fulltext indexer across three threads:
-
-```bash
-MAGE_INDEXER_THREADS_COUNT=3 php -f bin/magento indexer:reindex catalogsearch_fulltext
-```
-
-{:.bs-callout .bs-callout-info}
-Only use the environment variable in the indexer command. Do not save the variable to your environment or your Magento installs and updates may have errors.
--->
+Reindexing all indexers can take a long time for stores with large numbers of products, customers, categories, and promotional rules.
```
Comment (hguthrie) [inline guides/v2.3/config-guide/cli/config-cli-subcommands-index.md]: This is an interesting one. I see a section for 2.3.1 deleted. In fact, I opened a bug on this: https://jira.corp.magento.com/browse/MAGEDOC-4019

Comment (jeff-matthews) [reply guides/v2.3/config-guide/cli/config-cli-subcommands-index.md]: Good eye @hguthrie! The associated ticket for that change is still open (MAGETWO-92037), so I think we need to ignore it for now.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**